File: 1768-merge-strings-alternately/1768-merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let idx1 = 0; 
    let idx2 = 0;
    let stringBuilder = [];
    const L1 = word1.length; L2 = word2.length;
    
    while (idx1 < L1 & idx2 < L2){
        stringBuilder.push(word1[idx1]);
        stringBuilder.push(word2[idx2]);
        idx1 += 1;
        idx2 += 1;
    }
    
    if (idx1 < L1){
        return stringBuilder.join('') + word1.slice(idx1);
    }
    return stringBuilder.join('') + word2.slice(idx2);
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        res = []
        
        for w1, w2 in zip(word1, word2):
            res.append(w1)
            res.append(w2)
        
        if len(word2) < len(word1):
            return "".join(res) + word1[len(word2):]
        elif len(word1) < len(word2):
            return "".join(res) + word2[len(word1):]
        else:
            return "".join(res)